Update benchmarking README (#243)
neuralmagic/deepsparse: src/deepsparse/benchmark_model/README.md

<!--
Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing,
software distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
-->

# Benchmarking

> The command-line (CLI) tool `deepsparse.benchmark` is meant to simplify benchmarking and be the central codepath for benchmarking the engine on ONNX or SparseZoo models.

Reviewer: ```suggestion
The command-line (CLI) tool `deepsparse.benchmark` is meant to simplify benchmarking and be the central code path for benchmarking the engine on ONNX or SparseZoo models.
```

## Quickstart

> After you have installed `deepsparse` to your Python environment, the benchmark tool is available on your CLI. It is simple to run expressive model benchmarks on the DeepSparse Engine with minimal parameters.

Reviewer: ```suggestion
After you have installed `deepsparse` in your Python environment, the benchmark tool is available on your CLI. It is simple to run expressive model benchmarks on the DeepSparse Engine with minimal parameters.
```

> To run an sparse FP32 MobileNetV1 at batch size 16 for 10 seconds for throughput using 8 streams of requests:

Reviewer: ```suggestion
To run a sparse FP32 MobileNetV1 at batch size 16 for 10 seconds for throughput using 8 streams of requests:
```

The number of executions is a result of setting the time duration to run executions for, set by the `-t` argument in seconds.

### Synchronous (Single-stream) Scenario

Set by the `--scenario sync` argument, the goal metric is latency per batch (ms/batch). This scenario submits a single inference request at a time to the engine, recording the time taken for a request to return an output. This mimics an edge deployment scenario.

The latency value reported is the mean of all latencies recorded during the execution period for the given batch size.

### Asynchronous (Multi-stream) Scenario

Set by the `--scenario async` argument, the goal metric is throughput in items per second (i/s). This scenario submits `--num_streams` concurrent inference requests to the engine, recording the time taken for each request to return an output. This mimics a model server or bulk batch deployment scenario.

The throughput value reported comes from measuring the number of finished inferences within the execution time and the batch size.

## Usage

> In most cases good performance will be found in the default options so it can be as simple as running the command with a SparseZoo model stub or your local ONNX model.

Reviewer: ```suggestion
In most cases, good performance will be found in the default options so it can be as simple as running the command with a SparseZoo model stub or your local ONNX model.
```

```
usage: deepsparse.benchmark [-h] [-b BATCH_SIZE] [-shapes INPUT_SHAPES]
                            [-ncores NUM_CORES] [-s {async,sync}] [-t TIME]
                            [-nstreams NUM_STREAMS] [-pin {none,core,numa}]
                            [-q] [-x EXPORT_PATH]
                            model_path

Benchmark ONNX models in the DeepSparse Engine

positional arguments:
  model_path            Path to an ONNX model file or SparseZoo model stub

optional arguments:
  -h, --help            show this help message and exit
  -b BATCH_SIZE, --batch_size BATCH_SIZE
                        The batch size to run the analysis for. Must be
                        greater than 0
  -shapes INPUT_SHAPES, --input_shapes INPUT_SHAPES
                        Override the shapes of the inputs, i.e. -shapes
                        "[1,2,3],[4,5,6],[7,8,9]" results in input0=[1,2,3]
                        input1=[4,5,6] input2=[7,8,9]
  -ncores NUM_CORES, --num_cores NUM_CORES
                        The number of physical cores to run the analysis on,
                        defaults to all physical cores available on the system
  -s {async,sync}, --scenario {async,sync}

>                         Choose between using a sync/async scenarios. This is

Reviewer: ```suggestion
                        Choose between using sync/async scenarios. This is
```

                        similar to the single-stream/multi-stream scenarios.

>                         Default value is async.

Reviewer: ```suggestion
                        The default value is async.
```
